php/php.editor/tools/ASTPHP5Parser.cup

```diff
@@ -3181,85 +3285,18 @@ constant_modifiers ::=
     RESULT = null;
 :}
 
-| ppp_modifiers:modifier
-{:
-    RESULT = modifier;
-:}
-
-| final_modifier:modifier
+| non_empty_member_modifiers:modifier
 {:
     RESULT = modifier;
 :}
-
-| final_modifier:fModifier ppp_modifiers:pModifier
-{:
-    int result = 0;
-    result |= fModifier.intValue();
-    result |= pModifier.intValue();
-    RESULT = Integer.valueOf(result);
-:}
-
-| ppp_modifiers:pModifier final_modifier:fModifier
-{:
-    int result = 0;
-    result |= pModifier.intValue();
-    result |= fModifier.intValue();
-    RESULT = Integer.valueOf(result);
-:}
 ;
 
 variable_modifiers ::=
-ppp_modifiers:modifier
+non_empty_member_modifiers:modifier
```

Comment:
The parser allows valid combinations so far. However, it's possible to maintain it no longer...because there are many modifiers.
So, the parser allows invalid combinations now.
Instead, errors are handled via `ModifiersCheckHintError`.

Reply:
I see.